standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/model/MColumnDescriptor.java

```diff
@@ -29,23 +30,33 @@
  * A wrapper around a list of columns.
  */
 public class MColumnDescriptor {
-  private List<MFieldSchema> cols;
+  private List<MColumn> columns;
+  private long id;
 
   public MColumnDescriptor() {}
 
-  /**
-   *
-   * @param cols
-   */
   public MColumnDescriptor(List<MFieldSchema> cols) {
-    this.cols = cols;
+    columns = new ArrayList<>(cols.size());
+    for (MFieldSchema schema : cols) {
+      columns.add(new MColumn(schema.getName(), schema.getType(), schema.getComment()));
+    }
+  }
+
+  public List<MColumn> getColumns() {
```

Comment:
Could we come up with a better naming to distinguish 2: `getColumns()` & `getCols()`?
keep `getCols()`, but rename `getColumns()` to `getFields` or similar

Reply:
ack